pywinauto/controls/uia_controls.py

```diff
@@ -661,48 +663,186 @@ def get_item(self, row):
+    def item_by_index(self, idx):
+        """Find a menu item specified by the index"""
+        item = self.items()[idx]
+        return item
+
+    # -----------------------------------------------------------
+    @staticmethod
+    def _activate(item):
+        """Activate the specified item"""
+        if not item.is_active():
+            # self.actions.log("[DEBUG] Set focus on", tem.texts())
+            item.set_focus()
+        try:
+            item.expand()
+            # self.actions.log("[DEBUG] Expand ", item.texts())
+        except(NoPatternInterfaceError):
+            pass
+
+    # -----------------------------------------------------------
+    def _sub_item_by_text(self, menu, name, exact):
+        """Find a menu sub-item by the specified text"""
+        sub_item = None
+
+        if exact:
+            for i in menu.items():
+                if name == i.window_text():
+                    sub_item = i
+                    break
+        else:
+            items = []
+            texts = []
+            for i in menu.items():
+                items.append(i)
+                texts.append(i.window_text())
+            sub_item = findbestmatch.find_best_match(name, texts, items)
+
+        self._activate(sub_item)
+
+        return sub_item
+
+    # -----------------------------------------------------------
+    def _sub_item_by_idx(self, menu, idx):
+        """Find a menu sub-item by the specified index"""
+        sub_item = None
+        items = menu.items()
+        if items:
+            sub_item = items[idx]
+            self._activate(sub_item)
+        return sub_item
+
+    # -----------------------------------------------------------
+    def item_by_path(self, path, exact=False):
+        """Find a menu item specified by the path
+
+        The full path syntax is specified in:
+        :py:meth:`pywinauto.menuwrapper.Menu.get_menu_path`
+        """
+        # Get the path parts
+        part0, parts = path.split("->", 1)
+        part0 = part0.strip()
+        if len(part0) == 0:
+            raise IndexError()
+
+        # Find a top level menu item and select it. After selecting this item
+        # a new Menu control is created and placed on the dialog. It can be
+        # a direct child or a descendant.
+        # Sometimes we need to re-discover Menu again
+        try:
+            menu = None
+            if part0.startswith("#"):
+                menu = self._sub_item_by_idx(self, int(part0[1:]))
+            else:
+                menu = self._sub_item_by_text(self, part0, exact)
+
+            if not menu.items():
+                menu = self.top_level_parent().descendants(control_type="Menu")[0]
+
+            for cur_part in [p.strip() for p in parts.split("->")]:
+                if cur_part.startswith("#"):
+                    menu = self._sub_item_by_idx(menu, int(cur_part[1:]))
+                else:
+                    menu = self._sub_item_by_text(menu, cur_part, exact)
```

Comment:
Interesting. `menu_or_item` would be more clear to me, but it can be left as is for brevity.
Good job, @airelil. Thanks a lot!